integrations/lib/logger/logger.go

```diff
@@ -90,6 +98,86 @@ func Setup(conf Config) error {
 	return nil
 }
 
+// NewSLog builds a slog.Logger from the logger.Config.
+func (conf Config) NewSLog() (*slog.Logger, error) {
+	var w io.Writer
+	switch conf.Output {
+	case "":
+		w = logutils.NewSharedWriter(os.Stderr)
+	case "stderr", "error", "2":
+		w = logutils.NewSharedWriter(os.Stderr)
+	case "stdout", "out", "1":
+		w = logutils.NewSharedWriter(os.Stdout)
+	case teleport.Syslog:
+		w = os.Stderr
+		sw, err := utils.NewSyslogWriter()
+		if err != nil {
+			slog.Default().ErrorContext(context.Background(), "Failed to switch logging to syslog", "error", err)
+			break
+		}
+
+		// If syslog output has been configured and is supported by the operating system,
+		// then the shared writer is not needed because the syslog writer is already
+		// protected with a mutex.
+		w = sw
+	default:
+		// Assume this is a file path.
+		sharedWriter, err := logutils.NewFileSharedWriter(conf.Output, config.LogFileDefaultFlag, config.LogFileDefaultMode)
+		if err != nil {
+			return nil, trace.Wrap(err, "failed to init the log file shared writer")
+		}
+		w = logutils.NewWriterFinalizer[*logutils.FileSharedWriter](sharedWriter)
+		if err := sharedWriter.RunWatcherReopen(context.Background()); err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	level := new(slog.LevelVar)
+	switch strings.ToLower(conf.Severity) {
+	case "", "info":
+		level.Set(slog.LevelInfo)
+	case "err", "error":
+		level.Set(slog.LevelError)
+	case teleport.DebugLevel:
+		level.Set(slog.LevelDebug)
+	case "warn", "warning":
+		level.Set(slog.LevelWarn)
+	case "trace":
+		level.Set(logutils.TraceLevel)
+	default:
+		return nil, trace.BadParameter("unsupported logger severity: %q", conf.Severity)
+	}
+
+	configuredFields, err := logutils.ValidateFields(extraFields)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var slogLogger *slog.Logger
+	switch strings.ToLower(conf.Format) {
+	case "":
+		fallthrough // not set. defaults to 'text'
+	case "text":
+		enableColors := utils.IsTerminal(os.Stderr)
+		slogLogger = slog.New(logutils.NewSlogTextHandler(w, logutils.SlogTextHandlerConfig{
+			Level:            level,
+			EnableColors:     enableColors,
+			ConfiguredFields: configuredFields,
+		}))
+		slog.SetDefault(slogLogger)
+	case "json":
+		slogLogger = slog.New(logutils.NewSlogJSONHandler(w, logutils.SlogJSONHandlerConfig{
+			Level:            level,
+			ConfiguredFields: configuredFields,
+		}))
+		slog.SetDefault(slogLogger)
+	default:
+		return nil, trace.BadParameter("unsupported log output format : %q", conf.Format)
+	}
+
+	return slogLogger, nil
+}
+
```

Comment:
This logic is a port from the `config.applyLogConfig` logic. The original `applyLogConfig` function depends on `servicecfg.Config` and I did not want to refactor this part as well (the plugins don't have a `servicecfg.Config`.